### PR TITLE
[DNM-2.5.7] system charts for gke-operator

### PR DESCRIPTION
**Pull request by Klaven**
these are the system charts for the gke-operator

### CODE REVIEW

File: charts/rancher-gke-operator/1.0.500/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+name: rancher-gke-operator
+description: A Helm chart for provisioning GKE clusters
+home: https://github.com/rancher/gke-operator
+sources:
+  - "https://github.com/rancher/gke-operator"
+version: 1.0.500
```
Comment (thedadams) [inline charts/rancher-gke-operator/1.0.500/Chart.yaml:7]: The first version should probably be `1.0.000`.

File: charts/rancher-gke-operator/1.0.500/questions.yaml [hunk 2]
```diff
@@ -0,0 +1 @@
+rancher_min_version: 2.6.7-alpha1
```
Comment (thedadams) [inline charts/rancher-gke-operator/1.0.500/questions.yaml:1]: I believe these changes are going into `2.5.7`.

File: charts/rancher-gke-operator/1.0.500/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      ke.cattle.io/operator: gke
+  template:
+    metadata:
+      labels:
+        ke.cattle.io/operator: gke
+    spec:
+      serviceAccountName: gke-operator
+      containers:
+      - name: gke-operator
+        image: {{ template "system_default_registry" . }}{{ .Values.eksOperator.image.repository }}:{{ .Values.eksOperator.image.tag }}
```
Comment (thedadams) [inline charts/rancher-gke-operator/1.0.500/templates/deployment.yaml:19]:
```suggestion
        image: {{ template "system_default_registry" . }}{{ .Values.gkeOperator.image.repository }}:{{ .Values.gkeOperator.image.tag }}
```

File: charts/rancher-gke-operator/1.0.500/values.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+global:
+  systemDefaultRegistry: ""
+
+eksOperator:
+  image:
+    repository: rancher/gke-operator
+    tag: v1.0.5
```
Comment (thedadams) [inline charts/rancher-gke-operator/1.0.500/values.yaml:7]: Same, version should probably be `1.0.0`.

File: charts/rancher-gke-operator/1.0.500/values.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+global:
+  systemDefaultRegistry: ""
+
+eksOperator:
```
Comment (thedadams) [inline charts/rancher-gke-operator/1.0.500/values.yaml:4]:
```suggestion
gkeOperator:
```

**Review by thedadams — CHANGES_REQUESTED** [review]

File: charts/rancher-gke-operator/1.0.500/Chart.yaml [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+appVersion: v1.0.5
```
Comment (rmweir) [inline charts/rancher-gke-operator/1.0.500/Chart.yaml:8]: App version should correspond to the gke-operator image version.

File: charts/rancher-gke-operator/1.0.500/templates/clusterrole.yaml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: gke-operator
+  namespace: cattle-system
+rules:
+  - apiGroups: ['']
+    resources: ['secrets']
+    verbs: ['get', 'list', 'create', 'watch']
```
Comment (rmweir) [inline charts/rancher-gke-operator/1.0.500/templates/clusterrole.yaml:9]: Is 'create' necessary? I know it is used in eks-operator to communicate some info that would be needed to generate a kubeconfig but curious if something similar is definitely needed or not. I don't see a use of the secrets client in gke-operator at all, is that part just not implemented yet?

**Review by rmweir — CHANGES_REQUESTED** [review]

File: charts/rancher-gke-operator/1.0.500/Chart.yaml (hunk 1 above)
Comment (cmurphy) [reply charts/rancher-gke-operator/1.0.500/Chart.yaml:7]: done

File: charts/rancher-gke-operator/1.0.500/Chart.yaml (hunk 6 above)
Comment (cmurphy) [reply charts/rancher-gke-operator/1.0.500/Chart.yaml:8]: done

File: charts/rancher-gke-operator/1.0.500/questions.yaml (hunk 2 above)
Comment (cmurphy) [reply charts/rancher-gke-operator/1.0.500/questions.yaml:1]: done

File: charts/rancher-gke-operator/1.0.500/templates/clusterrole.yaml (hunk 7 above)
Comment (cmurphy) [reply charts/rancher-gke-operator/1.0.500/templates/clusterrole.yaml:9]: removed this for now

File: charts/rancher-gke-operator/1.0.500/templates/deployment.yaml (hunk 3 above)
Comment (cmurphy) [reply charts/rancher-gke-operator/1.0.500/templates/deployment.yaml:19]: done

File: charts/rancher-gke-operator/1.0.500/templates/gkeclusterconfig.yaml [hunk 8]
```diff
@@ -0,0 +1,286 @@
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (cmurphy) [inline charts/rancher-gke-operator/1.0.500/templates/gkeclusterconfig.yaml:1]: updated this

File: charts/rancher-gke-operator/1.0.500/values.yaml (hunk 5 above)
Comment (cmurphy) [reply charts/rancher-gke-operator/1.0.500/values.yaml:4]: done

File: charts/rancher-gke-operator/1.0.500/values.yaml (hunk 4 above)
Comment (cmurphy) [reply charts/rancher-gke-operator/1.0.500/values.yaml:7]: done

**Review by cmurphy — COMMENTED** [review]
Addressed feedback in https://github.com/rancher/system-charts/pull/420

**Comment by cmurphy** [comment]
Changed the version and appVersion again, also added back the secret perms including create in light of this comment https://github.com/rancher/gke-operator/pull/3#pullrequestreview-599809746

**Pull request closed without merging**